src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -106,3 +109,63 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 
     return nodejsLaunchConfig
 }
+
+/**
+ * Recursively searches directory and all nested directories until a TypeScript file is found
+ * @param dir Directory to search
+ * @returns true if at least one TypeScript file exists
+ */
+function hasTypeScriptFilesRecursive(dir:string): boolean {
```

Comment:
`filesystemUtilities.ts` has a similar function `isInDirectory()` but it is semantically not re-usable.

But `hasTypeScriptFilesRecursive()` is trivially generalizable so it does not belong in typescriptSamDebug.ts. If it's needed, it should live in  `filesystemUtilities.ts`. But I don't think it's needed, because we already depend on the `glob` package. See here for example usage: https://github.com/aws/aws-toolkit-vscode/blob/bab1f099477b59991645d051b17ef3bcbfe03baf/src/test/testRunner.ts#L67

Reply:
Done